File: sdk/src/resources/servers.ts
import { BaseResource } from './base';
import type {
  Server,
  ServerCreate,
  ServerUpdate,
  ServerAction,
  ServerStats,
  ServerBackup,
  ApiResponse,
  PaginatedResponse,
  PaginationParams,
} from '@escluse/types';

export class ServersResource extends BaseResource {
  async list(params?: PaginationParams & { status?: string; nodeId?: string }): Promise<PaginatedResponse<Server>> {
    const qs = this.buildQueryString(params);
    return this.request<PaginatedResponse<Server>>(`/servers${qs}`);
  }

  async get(serverId: string): Promise<ApiResponse<Server>> {
    return this.request<ApiResponse<Server>>(`/servers/${serverId}`);
  }

  async create(data: ServerCreate): Promise<ApiResponse<Server>> {
    return this.request<ApiResponse<Server>>('/servers', {
      method: 'POST',
      body: JSON.stringify(data),
    });
  }

  async update(serverId: string, data: ServerUpdate): Promise<ApiResponse<Server>> {
    return this.request<ApiResponse<Server>>(`/servers/${serverId}`, {
      method: 'PATCH',
      body: JSON.stringify(data),
    });
  }

  async delete(serverId: string): Promise<ApiResponse<void>> {
    return this.request<ApiResponse<void>>(`/servers/${serverId}`, {
      method: 'DELETE',
    });
  }

  async action(serverId: string, action: ServerAction): Promise<ApiResponse<void>> {
    return this.request<ApiResponse<void>>(`/servers/${serverId}/actions`, {
      method: 'POST',
      body: JSON.stringify(action),
    });
  }

  async start(serverId: string): Promise<ApiResponse<void>> {
    return this.action(serverId, { action: 'start' });
  }

  async stop(serverId: string): Promise<ApiResponse<void>> {
    return this.action(serverId, { action: 'stop' });
  }

  async restart(serverId: string): Promise<ApiResponse<void>> {
    return this.action(serverId, { action: 'restart' });
  }

  async kill(serverId: string): Promise<ApiResponse<void>> {
    return this.action(serverId, { action: 'kill' });
  }

  async reinstall(serverId: string): Promise<ApiResponse<void>> {
    return this.action(serverId, { action: 'reinstall' });
  }

  async stats(serverId: string): Promise<ApiResponse<ServerStats>> {
    return this.request<ApiResponse<ServerStats>>(`/servers/${serverId}/stats`);
  }

  async logs(serverId: string, params?: { lines?: number; search?: string }): Promise<ApiResponse<string[]>> {
    const qs = this.buildQueryString(params);
    return this.request<ApiResponse<string[]>>(`/servers/${serverId}/logs${qs}`);
  }

  async command(serverId: string, command: string): Promise<ApiResponse<void>> {
    return this.request<ApiResponse<void>>(`/servers/${serverId}/command`, {
      method: 'POST',
      body: JSON.stringify({ command }),
    });
  }

  async backups(serverId: string): Promise<ApiResponse<ServerBackup[]>> {
    return this.request<ApiResponse<ServerBackup[]>>(`/servers/${serverId}/backups`);
  }

  async createBackup(serverId: string, name?: string): Promise<ApiResponse<ServerBackup>> {
    return this.request<ApiResponse<ServerBackup>>(`/servers/${serverId}/backups`, {
      method: 'POST',
      body: JSON.stringify({ name }),
    });
  }

  async deleteBackup(serverId: string, backupId: string): Promise<ApiResponse<void>> {
    return this.request<ApiResponse<void>>(`/servers/${serverId}/backups/${backupId}`, {
      method: 'DELETE',
    });
  }
}

File: sdk-python/escluse/resources/servers.py
from typing import Dict, Any, Optional
from .base import BaseResource


class ServersResource(BaseResource):
    def list(
        self,
        page: int = 1,
        per_page: int = 20,
        status: Optional[str] = None,
        node_id: Optional[str] = None,
    ) -> Dict[str, Any]:
        params = {"page": page, "per_page": per_page}
        if status:
            params["status"] = status
        if node_id:
            params["nodeId"] = node_id
        return self._request("GET", "/servers", params=params)

    def get(self, server_id: str) -> Dict[str, Any]:
        return self._request("GET", f"/servers/{server_id}")

    def create(self, data: Dict[str, Any]) -> Dict[str, Any]:
        return self._request("POST", "/servers", json=data)

    def update(self, server_id: str, data: Dict[str, Any]) -> Dict[str, Any]:
        return self._request("PATCH", f"/servers/{server_id}", json=data)

    def delete(self, server_id: str) -> Dict[str, Any]:
        return self._request("DELETE", f"/servers/{server_id}")

    def start(self, server_id: str) -> Dict[str, Any]:
        return self._request(
            "POST", f"/servers/{server_id}/actions", json={"action": "start"}
        )

    def stop(self, server_id: str) -> Dict[str, Any]:
        return self._request(
            "POST", f"/servers/{server_id}/actions", json={"action": "stop"}
        )

    def restart(self, server_id: str) -> Dict[str, Any]:
        return self._request(
            "POST", f"/servers/{server_id}/actions", json={"action": "restart"}
        )

    def kill(self, server_id: str) -> Dict[str, Any]:
        return self._request(
            "POST", f"/servers/{server_id}/actions", json={"action": "kill"}
        )

    def reinstall(self, server_id: str) -> Dict[str, Any]:
        return self._request(
            "POST", f"/servers/{server_id}/actions", json={"action": "reinstall"}
        )

    def stats(self, server_id: str) -> Dict[str, Any]:
        return self._request("GET", f"/servers/{server_id}/stats")

    def logs(self, server_id: str, lines: int = 100) -> Dict[str, Any]:
        return self._request("GET", f"/servers/{server_id}/logs", params={"lines": lines})

    def command(self, server_id: str, command: str) -> Dict[str, Any]:
        return self._request(
            "POST", f"/servers/{server_id}/command", json={"command": command}
        )

    def backups(self, server_id: str) -> Dict[str, Any]:
        return self._request("GET", f"/servers/{server_id}/backups")

    def create_backup(self, server_id: str, name: Optional[str] = None) -> Dict[str, Any]:
        return self._request(
            "POST", f"/servers/{server_id}/backups", json={"name": name} if name else {}
        )

    def delete_backup(self, server_id: str, backup_id: str) -> Dict[str, Any]:
        return self._request("DELETE", f"/servers/{server_id}/backups/{backup_id}")